nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/src/main/java/org/apache/nifi/prometheus/util/PrometheusMetricsUtil.java

```diff
@@ -362,4 +365,149 @@ public static CollectorRegistry createBulletinMetrics(BulletinMetricsRegistry bu
+        if (percentUsed >= 100) {
+            aggregatedMetrics.put(atBackpressureKey, 1.0);
+        } else if (!aggregatedMetrics.containsKey(atBackpressureKey)) {
+            aggregatedMetrics.put(atBackpressureKey, 0.0);
+        }
+    }
+
+    private static void determineMinValueForPredictions(final Map<String, Double> aggregatedMetrics, final String metricFamilySamplesName,
+                                                        final double metricSampleValue, final String atBackpressureKey) {
+        final Double currentValue = aggregatedMetrics.get(metricFamilySamplesName);
+        if (aggregatedMetrics.get(atBackpressureKey) != null && aggregatedMetrics.get(atBackpressureKey) == 1.0) {
+            aggregatedMetrics.put(metricFamilySamplesName, 0.0);
+        } else if (currentValue == null) {
+            aggregatedMetrics.put(metricFamilySamplesName, metricSampleValue);
+        } else if (metricSampleValue > -1) {
+            if (currentValue == -1) {
+                aggregatedMetrics.put(metricFamilySamplesName, metricSampleValue);
+            } else {
+                aggregatedMetrics.put(metricFamilySamplesName, Math.min(metricSampleValue, currentValue));
+            }
+        }
+    }
+
+    private static void determineMaxValueForPercentUsed(final Map<String, Double> aggregatedMetrics, final String metricFamilySamplesName, final double metricSampleValue) {
+
+        Double currentValue = aggregatedMetrics.get(metricFamilySamplesName);
+        if (currentValue == null) {
+            currentValue = 0.0;
+        }
+        aggregatedMetrics.put(metricFamilySamplesName, Math.max(metricSampleValue, currentValue));
+    }
+
+    public static CollectorRegistry createAggregatedConnectionStatusAnalyticsMetrics(final ConnectionAnalyticsMetricsRegistry connectionAnalyticsMetricsRegistry,
+                                                                                     final Map<String, Double> aggregatedMetrics,
+                                                                                     final String instId, final String compType, final String compName, final String compId) {
+
+        final String instanceId = StringUtils.isEmpty(instId) ? DEFAULT_LABEL_STRING : instId;
+        final String componentType = StringUtils.isEmpty(compType) ? DEFAULT_LABEL_STRING : compType;
+        final String componentName = StringUtils.isEmpty(compName) ? DEFAULT_LABEL_STRING : compName;
+        final String componentId = StringUtils.isEmpty(compId) ? DEFAULT_LABEL_STRING : compId;
+        final Double bytesValue = aggregatedMetrics.get("nifi_time_to_bytes_backpressure_prediction");
+        final Double countsValue = aggregatedMetrics.get("nifi_time_to_count_backpressure_prediction");
+        final double bytesBackpressure = bytesValue == null ? -1.0 : bytesValue;
+        final double countsBackpressure = countsValue == null ? -1.0 : countsValue;
```

Comment:
Similar to the other comment, `-1.0` could be defined as `UNDEFINED_BACKPRESSURE`.